content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+
+<div class="release-feature-authors">authors: @NathanSWard, @MrGVSV</div>
+
+Bevy's reflection system now supports reflecting Rust arrays, which can be interacted with in a type-erased manner using the new [`Array`] trait. 
+
+```rust
+#[derive(Reflect)]
+struct Sprite {
+    position: [f32; 2],
+}
+
+let sprite = Sprite {
+    position: [0.1, 0.2],
+};
+
+let position = sprite.field("position").unwrap();
+if let ReflectRef::Array(array) = position.reflect_ref() {
+    let x = array.get(0).unwrap();
+    assert_eq!(x.downcast_ref::<f32>(), Some(&0.1));
+}
+```
+
+[`Array`]: https://docs.rs/bevy/0.8.0/bevy/reflect/trait.Array.html
+
+### Resource Reflection
+
+<div class="release-feature-authors">authors: @Shatur</div>
+
+Bevy ECS resources can now be reflected:
+
+```rust
+#[derive(Reflect)]
+#[reflect(Resource)]
+struct Scoreboard {
+    points: usize,
+}
+```
+
+This registers a [`ReflectResource`] in the [`TypeRegistry`] entry for the type, enabling type-erased read/write operations for the resource in an ECS [`World`]. 
+
+[`ReflectResource`]: https://docs.rs/bevy/0.8.0/bevy/ecs/reflect/struct.ReflectResource.html
+[`World`]: https://docs.rs/bevy/0.8.0/bevy/ecs/world/struct.World.html
+
+### Pretty Reflect Debug Formatting
+
+<div class="release-feature-authors">authors: @MrGVSV</div>
+
+"Debug printing" [`Reflect`] references now provides pretty / useful outputs.
+
+Consider the following example: 
+
+```rust
+#[derive(Reflect)]
+struct Foo {
+    a: f32,
+    b: Bar,
+}
+
+#[derive(Reflect)]
+struct Bar {
+    x: String,
+    y: usize,
+}
+let foo = Foo {
+    a: 42.0,
+    b: Bar {
+        x: "hello".to_string(),
+        y: 123,
+    },
+};
+
+let foo_reflect: &dyn Reflect = &foo;
+println!("{:#?}", foo_reflect);
+```
+
+In previous versions of Bevy, this would have printed:
+
+```
+Reflect(sprite::setup::Foo)
+```
+
+In **Bevy 0.8**, it prints:
+
+```
+sprite::setup::Foo {
+    a: 42.0,
+    b: sprite::setup::Bar {
+        x: "hello",
+        y: 123,
+    },
+}
+```
+
+Much better!
+
+### bevy_reflect Internal Refactors
+
+<div class="release-feature-authors">authors: @MrGVSV, @PROMETHIA-27, @jakobhellermann</div>
+
+Now that `bevy_reflect` is starting to get some serious investment and usage, we've invested time in reworking the internals to make them easier to maintain and extend:
+
+* **[`Reflect`] Derive Reorganization**: the derive logic was broken up into smaller, more maintainable pieces. "Metadata structs" were added to collect and organize derive inputs. (`@MrGVSV`) 
+* **The [`Reflect`] trait is now safe to implement**: Soundness no longer hinges on the implementor doing the right thing, thanks to some changes to the [`Reflect`] interface. As a result, we were able to remove the "unsafe" keyword from the [`Reflect`] trait. (`@PROMETHIA-27`)
+* "Serialize" logic is now implemented using [`TypeRegistry`] type data like other reflected trait logic, rather than being hard-coded into [`Reflect`] impls. (`@jakobhellermann`)
```

Comment:
```suggestion
* `Serialize` logic is now implemented using [`TypeRegistry`] type data like other reflected trait logic, rather than being hard-coded into [`Reflect`] impls. (`@jakobhellermann`)
```

Also maybe doing a [`reference`] (I assume this is what they are) would be good?